### DISCUSSION TITLE
node.js fails build front-core

### DISCUSSION THREAD

**Discussion by ebrookhouse72** [PrestaShop/PrestaShop#33319] [Q&A]
In trying to build 8, or this also is true for latest pull of develop, I'm getting an error from LoadRunner.js (I think) when attempting to do a basic installation. I Installed a few years ago, and recently upgraded the OS (Fedora 38), which allowed the existing PS install to keep working, but then I tried to upgrade to 8.0 and have run into this. 

Using this as my guide https://devdocs.prestashop-project.org/8/basics/installation/

If I make assets, I run into the issue, so I broke it down compiled individually to see that it was the front-core that was having the issue.

 make front-core
./tools/assets/build.sh front-core
>>> Building core theme assets...
/var/www/brantiques/themes /var/www/brantiques
npm WARN deprecated debug@4.1.1: Debug versions >=3.2.0 <3.2.7 || >=4 <4.3.1 have a low-severity ReDos regression when used in a Node.js environment. It is recommended you upgrade to 3.2.7 or 4.3.1. (https://github.com/visionmedia/debug/issues/797)
npm WARN deprecated debug@4.1.1: Debug versions >=3.2.0 <3.2.7 || >=4 <4.3.1 have a low-severity ReDos regression when used in a Node.js environment. It is recommended you upgrade to 3.2.7 or 4.3.1. (https://github.com/visionmedia/debug/issues/797)
npm WARN deprecated querystring@0.2.0: The querystring API is considered Legacy. new code should use the URLSearchParams API instead.
npm WARN deprecated babel-eslint@10.1.0: babel-eslint is now @babel/eslint-parser. This package will no longer receive updates.
npm WARN deprecated buffer@4.9.1: This version of 'buffer' is out-of-date. You must update to v4.9.2 or newer

added 482 packages, and audited 483 packages in 11s

42 packages are looking for funding
  run `npm fund` for details

11 vulnerabilities (3 moderate, 6 high, 2 critical)

To address all issues, run:
  npm audit fix

Run `npm audit` for details.

> prestashop-core-theme-js@1.7.5 build
> webpack --progress --mode production

10% building 0/1 entries 1/1 dependencies 0/1 modules/var/www/brantiques/themes/node_modules/loader-runner/lib/LoaderRunner.js:119
                        throw new Error("callback(): The callback was already called.");
                              ^

Error: callback(): The callback was already called.
    at context.callback (/var/www/brantiques/themes/node_modules/loader-runner/lib/LoaderRunner.js:119:10)
    at Object.ESBuildLoader (/var/www/brantiques/themes/node_modules/esbuild-loader/dist/loader.js:62:9)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)

Node.js v18.16.0
make: *** [Makefile:11: front-core] Error 1

This is PHP 8.2.7
mysqlnd 8.2.7

All PHP configuration and Extensions are green (referring to PHP PrestaShop Info)

Should I give up on PS for a few months? Should I downgrade my OS (PHP maybe Node.js)?

Any advice appreciated.

Edward

**Comment by kpodemski** [comment]
Hi

For the front office, it's best to use Node 16, for back office 14.
